app/models/credit_note.rb

```diff
@@ -181,53 +186,64 @@ def ensure_number
 
     self.number = "#{invoice.number}-CN#{formatted_sequential_id}"
   end
+
+  def ensure_metadata_consistency
+    return unless metadata
+
+    if metadata.organization_id != organization_id
+      errors.add(:metadata, "must belong to the same organization as the credit note")
+    end
+  end
```

Comment:
I'm not sure about this kind of verification. 
Now, if you change the credit note status, the app will load the metadata to ensure it's organization_id is correct.
Maybe we could do this kind of check on creation only. This reduce the risk of a mistake but keep things light.

Today, we don't have this kind of validation anywhere in the app. I feel like if we start adding this all over the place it's going to trigger a lot of extra requests.

Please share your thoughts!

Reply:
I gave this more thoughts and I think it should be done in the services instead. You'd usually have all the model and ids necessary for this type of checks.